platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/assertions/SingleNodeMetricsAssert.java

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.otter.fixtures.assertions;
+
+// SPDX-License-Identifier: Apache-2.0
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import org.assertj.core.api.AbstractAssert;
+import org.hiero.otter.fixtures.result.SingleNodeMetricsResult;
+
+/**
+ * Assertion class for {@link SingleNodeMetricsResult}.
+ *
+ * <p>Provides custom assertions for validating metric results for one node.
+ */
+@SuppressWarnings({"unused", "UnusedReturnValue"})
+public class SingleNodeMetricsAssert extends AbstractAssert<SingleNodeMetricsAssert, SingleNodeMetricsResult> {
+
+    /**
+     * Constructs an assertion for the given {@link SingleNodeMetricsResult}.
+     *
+     * @param actual the actual {@link SingleNodeMetricsResult} to assert
+     */
+    protected SingleNodeMetricsAssert(@Nullable final SingleNodeMetricsResult actual) {
+        super(actual, SingleNodeMetricsAssert.class);
+    }
+
+    /**
+     * Creates an assertion for the given {@link SingleNodeMetricsResult}.
+     *
+     * @param actual the actual {@link SingleNodeMetricsResult} to assert
+     * @return a new instance of {@link SingleNodeMetricsResult}
+     */
+    public static SingleNodeMetricsAssert assertThat(@Nullable final SingleNodeMetricsResult actual) {
+        return new SingleNodeMetricsAssert(actual);
+    }
+
+    /**
+     * Verifies that the specified metric value was at no point higher than the given threshold.
+     *
+     * @param threshold the exclusive upper bound
+     * @return the {@link SingleNodeMetricsResult} instance
+     */
+    public SingleNodeMetricsAssert neverExceeds(final double threshold) {
+        isNotNull();
+        final boolean exceeded = actual.history().stream()
+                .mapToDouble(SingleNodeMetricsAssert::toDouble)
+                .anyMatch(v -> v > threshold);
+        if (exceeded) {
+            final List<Double> wrongValues = actual.history().stream()
+                    .map(SingleNodeMetricsAssert::toDouble)
+                    .filter(v -> v > threshold)
+                    .toList();
+            failWithMessage(
```

Comment:
`wrongValues` can potentially be very long. I wonder if there is a better way to report this. Maybe the maximum would be sufficient?

We should also report the timestamp when the limit was exceeded.